server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java

```diff
@@ -0,0 +1,400 @@
+                }
+
+                if (!resteasyReq.getAsyncContext().isSuspended()) {
+                   try {
+                      resteasyResp.close();
+                   } catch (IOException e) {
+                      throw new RuntimeException(e);
+                   }
+                }
+
+                final Mono<Void> actualMono = Optional.ofNullable(resteasyReq.timeout())
+                    .map(timeout -> {
+                       isTimeoutSet.set(true);
+                       return completionSink.asMono().timeout(resteasyReq.timeout());
+                    })
+                    .orElse(completionSink.asMono());
+
+                return actualMono
+                    .doFinally(s -> {
+                       try {
+                          body.close();
+                       } catch (final IOException ioe) {
+                          log.error("Failure to close the request's input stream.", ioe);
+                       }
+                    });
+             }).onErrorResume(t -> {
+                if (!resteasyResp.isCommitted()) {
+                   final Mono<Void> sendMono;
+
+                   if (isTimeoutSet.get() && Exceptions.unwrap(t) instanceof TimeoutException) {
+                      sendMono = resp.status(503).send();
+                   } else {
+                      log.error("Unhandled server error.", t);
+                      sendMono = resp.status(500).send();
+                   }
+                   SinkSubscriber.subscribe(completionSink, sendMono);
+                } else {
+                   log.error("Unhandled server error, JAXRS response committed.", t);
+                }
+
+                return completionSink.asMono();
+             });
+      }
+
+   }
+
+   @Override
+   public void stop()
+   {
+      runtimePort = -1;
+      server.disposeNow();
+      if (deployment != null) {
+         deployment.stop();
+      }
+   }
+
+   @Override
+   public ResteasyDeployment getDeployment() {
+      if (deployment == null)
+      {
+         deployment = new ResteasyDeploymentImpl();
+      }
+      return deployment;
+   }
+
+   @Override
+   public ReactorNettyJaxrsServer setDeployment(ResteasyDeployment deployment)
+   {
+      this.deployment = deployment;
+      return this;
+   }
+
+   @Override
+   public ReactorNettyJaxrsServer setPort(int port) {
+      this.configuredPort = port;
+      return this;
+   }
+
+   public int getPort() {
+      return runtimePort > 0 ? runtimePort : configuredPort;
+   }
+
+   @Override
+   public ReactorNettyJaxrsServer setHostname(String hostname) {
+      this.hostname = hostname;
+      return this;
+   }
+
+   public String getHostname() {
+      return hostname;
+   }
+
+   @Override
+   public ReactorNettyJaxrsServer setRootResourcePath(String rootResourcePath)
+   {
+      root = rootResourcePath;
+      if (root != null && root.equals("/")) {
+         root = "";
+      } else if (!root.startsWith("/")) {
```

Comment:
This could produce an NPE if `rootResourcePath` is `null`.

Reply:
What would you like it to throw?  I generally prefer to bubble NPE.  However, I would likely have used `Objects#requireNonNull` on line 87.  Should it set it to `null` or the empty string when the supplied argument is null?  I will check the javadoc to see if it gives guidance.

Reply:
IMO an NPE here would be a bug. Unfortunately there doesn't seem to be any JavaDoc's so we don't have much to go on. However the fact that the first if checks for not null makes me assume null is allowed. So maybe the first like should be:
```
if (root == null || root.equals("/") {
    root = ""
} else ...
```

Reply:
@ronsigal Do you know if a `null` root resource is valid?

Reply:
Ah fair point about this being a bug.  I know you asked @ronsigal about it, but minimally the code should be better/clearer.  I will make a tweak.